src/amber/cli/recipes/recipe_fetcher.cr

```diff
@@ -0,0 +1,70 @@
+require "http/client"
+require "zip"
+
+module Amber::Recipes
+  class RecipeFetcher
+    getter kind : String
+    getter name : String
+    getter directory : String
+
+    def initialize(@kind : String, @name : String)
+      @directory = "#{__DIR__}/#{@name}/#{@kind}"
+    end
+
+    def fetch
+      # if the recipe exists in the local recipes folder then use that
+      if Dir.exists?(@directory)
+        return @directory
+      end
+
+      # if the given recipe name is a directory and has a sub directory
+      # for the kind of component being generated then use it
+      if Dir.exists?("#{@name}/#{@kind}")
+        return "#{@name}/#{@kind}"
+      end
+
+      # check the cached templates folder
+      template_path = "#{AMBER_RECIPE_FOLDER}/#{@name}"
+
+      if Dir.exists?("#{template_path}/#{@kind}")
+        return "#{template_path}/#{@kind}"
+      end
+
+      # otherwise fetch from the github repository
+      return fetch_url template_path
+    end
+
+    def fetch_url(template_path)
+      # download the recipe zip file from the github repository
+      HTTP::Client.get("https://raw.githubusercontent.com/AmberRecipes/recipes/master/#{@name}.zip") do |response|
+        if response.status_code != 200
+          # raise an exception if the recipe zip was not found
+          raise "Could not find that recipe"
```

Comment:
Should we print a full exception here?

See #686 and #671 (Additional Information)